src/alire/alire-origins-deployers-system-rh_rpm_wrapper.adb

```diff
@@ -0,0 +1,125 @@
+with Alire.Errors;
+
+with GNAT.Regpat;
+
+package body Alire.Origins.Deployers.System.Rh_Rpm_Wrapper is
+
+   package Subprocess renames Alire.OS_Lib.Subprocess;
+
+   ---------------------
+   -- Wrapper_Command --
+   ---------------------
+
+   function Wrapper_Command (This : Deployer) return String is
+     (case This.Wrapper is
+         when Dnf => "dnf",
+         when Yum => "yum");
+
+   ----------------------
+   -- Already_Installed --
+   -----------------------
+
+   overriding
+   function Already_Installed (This : Deployer) return Boolean is
+      Pck    : String renames This.Base.Package_Name;
+
+      Wrapper : constant String := Wrapper_Command (This);
+
+      Output : constant Utils.String_Vector :=
+                 Subprocess.Checked_Spawn_And_Capture
+                   (Wrapper,
+                    Empty_Vector &
+                      "-y" &
+                      "list" &
+                      "--installed" &
+                      This.Base.Package_Name,
+                    Valid_Exit_Codes => (0, 1), -- Returned when not found
+                    Err_To_Out     => True);
+   begin
+      if not Output.Is_Empty
+        and then
+          Output.First_Element = Pck
+      then
+         return True;
+      else
+         Trace.Detail ("Cannot find package '" & Pck &
+                         "' in " & Wrapper & " installed list: '" &
+                         Output.Flatten & "'");
+         return False;
+      end if;
+   end Already_Installed;
+
+   ------------
+   -- Detect --
+   ------------
+
+   overriding
+   function Detect (This : Deployer) return Version_Outcomes.Outcome is
+
+      Regexp : constant String := "^" & This.Base.Package_Name &
+        "[^\s]+\s+([^\s]+)";
+
+      Wrapper : constant String := Wrapper_Command (This);
+
+      Output    : constant Utils.String_Vector :=
+        Subprocess.Checked_Spawn_And_Capture
+          (Wrapper,
+           Empty_Vector &
+             "-y" &
+             "list" &
+             This.Base.Package_Name,
+           Valid_Exit_Codes => (0, 1), -- Returned when not found
+           Err_To_Out       => True);
+
+      use GNAT.Regpat;
+      Matches : Match_Array (1 .. 1);
+   begin
+      for Line of Output loop
+         Trace.Debug ("Extracting native version from " & Wrapper &
+                        " output: " & Line);
+         Match (Regexp, Line, Matches);
+         if Matches (1) /= No_Match then
+            Trace.Debug ("Candidate version string: "
+                         & Line (Matches (1).First .. Matches (1).Last));
+            return
+              Version_Outcomes.New_Result
+                (Semantic_Versioning.Parse
+                   (Line (Matches (1).First .. Matches (1).Last),
+                    Relaxed => True)); --  Relaxed because some Fedora
+                                       --  versions have extra version pts,
+                                       --  e.g. 5.8-3.fc33
```

Comment:
According to semantic versioning, `5.8-3 < 5.8` (dash signals a pre-release). If Fedora instead uses it to provide custom info for true `5.8` versions, it should be parsed as `5.8+3`, for example.

Reply:
A bit overkill but Rh_To_Semver function should do the trick.

Reply:
Consult [Fedora Package Versioning Examples](https://fedoraproject.org/wiki/Package_Versioning_Examples) : there are a lot of possibilities !
In Redhat there are even sometimes 4 digits to identify a version !

Reply:
Yup, same happens with debians (https://github.com/alire-project/alire/blob/906095fcb3f018331c46adef6ab2df391acffbcf/src/alire/alire-origins-deployers-system-apt.adb#L44). Fortunately they provide a convenient regexp. But, bottom line, the important thing is that RH does not use "-" in the Semver sense, so we cannot simply parse their versions as-is, so thanks for taking care of that.

Reply:
That's what I did with commit 723da69.

Would you like to review newly added Rh_To_Semver function ?

Reply:
Right, I saw it, I haven't had the time to review it yet :)